concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -3673,6 +3696,26 @@ migrateBlockPointers migration BlockStatePointers{..} = do
               bspRewardDetails = newRewardDetails
             }
 
+-- | Thaw the block state, making it ready for modification.
+--  This function wraps the underlying 'PersistentBlockState' of the provided 'HasedPersistentBlockState' in a new 'IORef'
+--  such that changes to the thawed block state does not propagate into the parent state.
+--
+--  Further the 'DiffMap.DifferenceMap' of the accounts structure in the provided block state is
+--  "bumped" in the sense that a new one is created for the new thawed block with a pointer to the parent difference map.
+--  The parent difference map is empty if the parent is persisted otherwise it may contain new accounts created in that block.
+doThawBlockState ::
+    (SupportsPersistentState pv m) =>
+    HashedPersistentBlockState pv ->
+    m (PersistentBlockState pv)
+doThawBlockState HashedPersistentBlockState{..} = do
+    -- This load is cheap as the underlying block state is retained in memory as we're building from it, so it must be the "best" block.
```

Comment:
The comment about invokeInstance is irrelevant. The whole state is not thawed there, my bad, only the contract state is.

However the other point about this being an odd place remains. In particular this is changing the preconditions of this method. This is a method on the BlockStateStorage class, which as of now does not say anything about best block.

It's fine to refine that, but please state these additional conditions in the typeclass definition, rather than inside the implementaion of the instance.

Reply:
I've added it as a precondition on the type class definition.